types/lodash/index.d.ts

```diff
@@ -6366,98 +6366,34 @@ declare namespace _ {
         /**
          * @see _.forEach
          */
-        each<T>(
-            collection: T[],
-            iteratee?: ListIterator<T, any>
-        ): T[];
-
-        /**
-         * @see _.forEach
-         */
-        each(
-            collection: string,
+        each<TString extends string | null | undefined>(
+            collection: TString,
             iteratee?: ListIterator<string, any>
-        ): string;
+        ): TString;
 
         /**
          * @see _.forEach
          */
-        each<T>(
-            collection: List<T>,
+        each<T, TList extends List<T> | null | undefined>(
```

Comment:
I just did some more testing, and this approach breaks type inference for the `ListIterator`, so this won't work property:

```
_.each([1,2,3], (value, index, collection) => {
   // value has type {}, but should be number
   // collection has type List<{}>, but should be number[]
});
```

Even my original change - `each<T, TList extends List<T> | null | undefined>(` - breaks the type inference. So I guess I'll revert it.